elyra/pipeline/pipeline.py

```diff
@@ -70,9 +70,9 @@ def __init__(self, id, type, classifier, filename, runtime_image, memory=None, c
         self._inputs = inputs or []
         self._outputs = outputs or []
         self._parent_operations = parent_operations or []
-        self._cpu = cpu
-        self._gpu = gpu
-        self._memory = memory
+        self._cpu = self.validate_resource_value("CPU", cpu)
+        self._gpu = self.validate_resource_value("GPU", gpu)
```

Comment:
Regarding my earlier comment relative to the front-end, it seems like we should tolerate a zero value for GPU.  Sorry, this complicates the check and messaging.